packages/forms/src/UIForm/fields/Datalist/Datalist.component.js

```diff
@@ -25,7 +49,57 @@ class Datalist extends Component {
 		this.props.onFinish(event, payloadWithSchema);
 	}
 
+	onFocus(event) {
+		if (this.props.schema.triggers) {
+			this.props.schema.triggers.forEach(trigger => {
+				if (trigger.onEvent === 'focus') {
+					this.setState({ isLoading: true });
+					this.props.onTrigger(event, { trigger }).then(data => {
+						this.setState({
+							isLoading: false,
+							...data,
+						});
+					});
+				}
+			});
+		}
+	}
+
+	getTitleMap() {
+		let titleMap = [];
+		if (this.state.titleMap) {
+			titleMap = Array.from(this.state.titleMap);
+		} else if (this.props.schema.titleMap) {
+			titleMap = Array.from(this.props.schema.titleMap);
+		}
+		const isMultiple = this.isMultiple();
+		if (this.props.value && !isMultiple) {
+			if (!titleMap.find(option => option.value === this.props.value)) {
+				titleMap.push({ name: this.props.value, value: this.props.value });
+			}
+		} else if (this.props.value && isMultiple) {
+			this.props.value.forEach(value => {
+				if (!titleMap.find(option => option.value === value)) {
+					titleMap.push({ name: this.props.value, value: this.props.value });
```

Comment:
Mistake here `{ name: value, value }`

But what do you think about this implem that avoid the duplicate LOC
```
getTitleMap() {
	let titleMap;
	if (this.state.titleMap) {
		titleMap = this.state.titleMap;
	} else if (this.props.schema.titleMap) {
		titleMap = this.props.schema.titleMap;
	} else {
		titleMap = [];
	}

	const isMultiple = this.isMultiple();
	const values = this.isMultiple() ? this.props.value : [this.props.value];
	const optionNotExists = value => titleMap.find(option => option.value === value);
	const additionalOptions = values.filter(optionNotExists).reduce((acc, value) => {
		acc.push({ name: value, value });
		return acc;
	}, []);

	return titleMap.concat(additionalOptions);
}
```